widgetschema.json

```diff
@@ -0,0 +1,244 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "WidgetProperties",
+    "type": "object",
+    "required": ["id", "name", "description", "version", "properties"],
+    "additionalProperties": false,
+    "properties": {
+        "id": {
+            "type": "string",
+            "description": "Unique identifier for the widget"
+        },
+        "name": {
+            "type": "string",
+            "description": "Name of the widget"
+        },
+        "description": {
+            "type": "string",
+            "description": "Description of the widget"
+        },
+        "version": {
+            "type": "string",
+            "description": "Version of the widget"
+        },
+        "src": {
+            "type": "string",
+            "description": "URL of the widget source code"
+        },
+        "needsDataSource": {
+            "type": "boolean",
+            "description": "Indicates if the widget needs a data source"
+        },
+        "needsInstance": {
+            "oneOf": [{ "type": "boolean" }, { "$ref": "#/definitions/dataType" }]
+        },
+        "typesSupported": {
+            "type": "array",
+            "items": { "$ref": "#/definitions/dataType" }
+        },
+        "needsDocument": {
+            "type": "string"
+        },
+        "icon": {
+            "type": "object",
+            "required": ["src"],
+            "additionalProperties": false,
+            "properties": {
+                "src": {
+                    "type": "string"
+                }
+            }
+        },
+        "colspan": {
+            "type": "integer",
+            "minimum": 1,
+            "maximum": 12
+        },
+        "allowSticky": {
+            "type": "boolean"
+        },
+        "deprecated": {
+            "type": "boolean"
+        },
+        "properties": {
+            "type": "array",
+            "uniqueItems": true,
+            "description": "Array of properties that the widget supports",
+            "items": { "$ref": "#/definitions/widgetProperty" }
+        },
+        "$schema": { "type": "string" }
+    },
+
+    "definitions": {
+        "dataType": {
+            "enum": ["documents", "objectInstances", "documentVersions", "relatedObjectInstances"]
+        },
+        "widgetSettingOptionalValue": {
```

Comment:
Is this still being used?  I thought we did away with this.

Reply:
The List Table, Page Header, and Correspondences use it

Reply:
@bradwray Can you clarify if this is still needed?  Last I spoke to Richa, she said this wasn't really needed anymore.

Reply:
I don't see this being used in any widgets anymore. @smark-1 you do?

Reply:
@actong, @smark-1 is right, there are several places that still have this configuration. Right now it is the way we are able to populate an initial value into a optional setting when the toggle is first toggled on.

Reply:
Does `initialValue` not already do that?  `isOptional.initialValue` seems redundant, not sure that it does anything that `initialValue` doesn't already do.

Reply:
Not sure. Since there are places where it is being used, should I make a tech debt task to remove the use of isOptional.initialValue and add that changes to this schema file will be needed too?

Reply:
I think if there's no real reason for `isOptional.initialValue` (in the sense everything can be handled by having `isOptional: boolean` and `initialValue`), then I wouldn't even bother including it in this schema.  If that results in the platform widgets getting warnings, then so be it, it'll give an incentive to update them, but I wouldn't want to include it in the schema and potentially encourage custom widgets to start using `widgetSettingOptionalValue`.